### ISSUE TITLE
Why use named placeholders when you can use anonymous?

### ISSUE THREAD

**Issue by Andrews54757** [catfan/Medoo#613]
https://github.com/catfan/Medoo/blob/master/src/Medoo.php#L351.

Why use named placeholders (EG ``SELECT * FROM `table` WHERE `arg` = :placeholder``) instead of anonymous ones? (``SELECT * FROM `table` WHERE `arg` = ?``)

The guid is being incremented anyway. I think it would be best practice to use unnamed placeholders for clarity and performance, as well as compactness.

I think there needs to be fixes to bottlenecks such as this. With a site with large traffic, the server can have trouble. 

Someone fancy for a PR?

**Comment by Yousha** [comment]
"anonymous" ? what's that?
You meant "positional parameter" ?

And the difference between  **positional parameter**(?) and **named parameters**(:) is that **named parameters** are a bit advanced, they don't have to be bound in order they appear. You can also repeat one named parameter multiple times in the statement, but you can bind it only once to pass the value.
^ this also leads you to write less code!

Also in **positional parameter**(?) you'll have to take care about the order in which they will be bound to the query.


**This issue must be closed.**

**Comment by Andrews54757** [comment]
@Yousha Positional parameters do not have to be bound in order.

http://php.net/manual/en/pdostatement.bindparam.php

**Comment by catfan** [comment]
@Andrews54757 I benchmarked (10000 times, with 5 placeholders in one query) the performance of positional parameter (14.7s ~ 14.9s) and named parameters (14.9s ~ 15.2s). There is no great performance difference between them. The positional parameter is only a little little bit faster.

The reason why we took named parameters is we need to maintain the key name and value type for the statement, and better for output the executed query for `debug()` and `log()`.

The pull request #617 is made a bug for this. For example, this will error when the input value contained with question mark. Confusing.
```php
$data = $database->debug()->select('account', [
	'user_name',
	'user_id'
], [
	'user_name' => 'foo?',
	'email' => 'bar?'
]);

// SELECT "user_name","user_id" FROM "account" WHERE "user_name" = 'foo'bar?'' AND "email" = ?
```

So, we have no plan and didn't see any great advanced to change it as positional parameter.

But this issue is open to discuss the benefit between this two ways.
